wit/preview2/redis.wit

```diff
@@ -0,0 +1,55 @@
+interface redis {
+  use fermyon:spin/redis-types.{payload, redis-parameter, redis-result}
+
+  /// Errors related to interacting with Redis
+  variant error {
+      /// An invalid address string
+      invalid-address,
+      /// There are too many open connections
+      too-many-connections,
+      /// A retrieved value was not of the correct type
+      type-error,
+      /// An I/O error occurred
+      io(string),
```

Comment:
Should we just call this `other(string)` or something? It seems unlikely that all our surprise errors would be IO-related...

(I don't really care for Redis in particular, but for (re)establishing a precedent)

Reply:
Yeah I suspect that pretty much any time we'd want another specific case we could be more precise than `io` :shrug: